x/gamm/keeper/pool_service_test.go

```diff
@@ -222,6 +242,181 @@ func (suite *KeeperTestSuite) TestCreateBalancerPool() {
 	}
 }
 
+func (suite *KeeperTestSuite) TestInitializePool() {
+	params := suite.App.GAMMKeeper.GetParams(suite.Ctx)
+	testAccount := suite.TestAccs[0]
+
+	// get raw pool creation fee(s) as DecCoins
+	poolCreationFeeDecCoins := sdk.DecCoins{}
+	for _, coin := range params.PoolCreationFee {
+		poolCreationFeeDecCoins = poolCreationFeeDecCoins.Add(sdk.NewDecCoin(coin.Denom, coin.Amount))
+	}
+
+	tests := []struct {
+		name        string
+		msg         swaproutertypes.CreatePoolMsg
+		usingMock   bool
+		emptySender bool
+		expectPass  bool
+	}{
+		{
+			name:        "initialize balancer pool with default assets",
+			msg:         balancer.NewMsgCreateBalancerPool(testAccount, defaultPoolParams, defaultPoolAssets, defaultFutureGovernor),
+			emptySender: false,
+			expectPass:  true,
+		},
+		{
+			name:        "initialize stableswap pool with default assets",
+			msg:         stableswap.NewMsgCreateStableswapPool(testAccount, defaultStableSwapPoolParams, defaultStableSwapPoolAssets, defaultScalingFactor, ""),
+			emptySender: false,
+			expectPass:  true,
+		},
+		{
+			name:      "initialize a CL pool which cause panic",
+			msg:       balancer.NewMsgCreateBalancerPool(testAccount, defaultPoolParams, defaultPoolAssets, defaultFutureGovernor),
+			usingMock: true,
+		},
+	}
+
+	for _, test := range tests {
```

Comment:
This is to make sure that there are no problems stemming from `test` of the wrong iteration being used. This is a Go-specific problem
```suggestion
	for _, test := range tests {
	    test := test
```